### DISCUSSION TITLE
Multisite plugin duplicates the site collection causing wrong information rendered

### DISCUSSION THREAD

**Discussion by isyedakhtar** [Sitecore/jss#1623] [Help] [answered]
### Description

The issue I noticed is related to xm foundation repo https://github.com/sitecorelabs/xmcloud-foundation-head. We can set the pos and host names on the SiteGrouping. In multisite environment, the plugin fetches the information of all the configured sites and created config.js which is then used at the build and run time. In addition to this, we can also configure one website name as jssAppName variable. During the build, the jssAppName is started as the first website and then all the sites collection from XMC is pulled by the multisite plugin. 
This plugin https://github.com/sitecorelabs/xmcloud-foundation-head/blob/main/src/sxastarter/src/lib/site-resolver/plugins/multisite.ts would then merge the two sets - jssAppName and collection from XMC. The resultant list is then used at the build and run time. If the jssAppName is same as one of the sites defined in XMC, that site gets duplicated in the list which can contain inconsistent information. I have seen the errors related to pos not defined etc. due to this duplication. 

```ts
class MultisitePlugin implements SiteResolverPlugin {
  exec(sites: SiteInfo[]): SiteInfo[] {
    // Add preloaded sites
    **sites.push(.**..(JSON.parse(config.sites) as SiteInfo[]));

    return sites;
  }
}
```

Artem suggested to keep the jssAppName empty which works for some of the scenarios. However, I noticed that doing this will cause issues in other places. For example, the error pages are using this config(404.tsx). 
To resolve this, I think the multisite.ts plugin in xm foundation should not merge the two sets but just replace the one it has built using the jssAppName. This is one of the potential fixes that I have tested. Thought to ask the group, what else could be a better solution.

**Comment by illiakovalenko** [comment, marked as answer]
@isyedakhtar Essentially, this problem happens because of [default](https://github.com/sitecorelabs/xmcloud-foundation-head/blob/main/src/sxastarter/src/lib/site-resolver/plugins/default.ts#L15) plugin, did you try to modify that? Default plugin adds default site besides the XMCloud configuration. The default configuration uses _*_, so this config is less specific.

**Reply by isyedakhtar** [reply]
Thanks @illiakovalenko  I modified the behaviour and instead of using push, I am replacing the sites collection with the one that multisite plugin loads in config. Since, we are expecting the preloaded sites to be part of this collection from multisite plugin, we should be fine.
